### PR TITLE
remove legacy Version classes, use new one

### PR DESCRIPTION
**Pull request by memsharded**
The ``Version`` class itself is not modified, simply taken from the ``recipe_ref.py`` and put it ``version.py``, overwriting what was there before.

Next steps:
- Remove the dependency to node-semver, and use our own VersionRange
- Implement a ``RequireReference`` model that can contain a VersionRange

### CODE REVIEW

File: conans/client/tools/oss.py [hunk 1]
```diff
@@ -202,69 +202,69 @@ class _OSVERSIONINFOEXW(ctypes.Structure):
...
+        elif version.major == 6:
             return "squeeze"
-        elif version.major() == "5.Y.Z":
+        elif version.major == 5:
             return "lenny"
-        elif version.major() == "4.Y.Z":
+        elif version.major == 4:
             return "etch"
-        elif version.minor() == "3.1.Z":
+        elif version.minor == 1:
```
Comment (madebr) [inline conans/client/tools/oss.py]: Don't you want to compare major version as well?
```suggestion
        elif (version.major, version.minor) == (3, 1):
```

File: conans/client/tools/oss.py [hunk 2]
```diff
@@ -202,69 +202,69 @@ class _OSVERSIONINFOEXW(ctypes.Structure):
...
             return "sarge"
-        elif version.minor() == "3.0.Z":
+        elif version.minor == 0:
```
Comment (madebr) [inline conans/client/tools/oss.py]:
```suggestion
        elif (version.major, version.minor) == (3, 0):
```

File: conans/client/tools/oss.py [hunk 3]
```diff
@@ -202,69 +202,69 @@ class _OSVERSIONINFOEXW(ctypes.Structure):
...
             return "Windows 7"
-        elif version.minor() == "6.2.Z":
+        elif version.major == 6 and version.minor == 2:
             return "Windows 8"
-        elif version.minor() == "6.3.Z":
+        elif version.major == 6 and version.minor == 3:
             return "Windows 8.1"
-        elif version.minor() == "10.0.Z":
+        elif version.major == 10:
             return "Windows 10"
```
Comment (madebr) [inline conans/client/tools/oss.py:235]:
```suggestion
        elif version.major == 10:
            return "Windows 10"
        elif version.major == 11:
            return "Windows 11"
```
(or perhaps these suggestions do not belong in this pr)

**Comment by memsharded** [comment]
Yeah, thanks for the suggestions, they are ok. I was just doing some quick migration to the new Version, and took the shortest path that didn't break, but it is certainly better to check majors too, just in case.

File: conans/client/tools/oss.py [hunk 4]
```diff
@@ -202,69 +202,71 @@ class _OSVERSIONINFOEXW(ctypes.Structure):
...
+        elif version.major == 11:
```
Comment (madebr) [inline conans/client/tools/oss.py:236]: Looking deeper into this, I'm not sure about this anymore.

Comment (memsharded) [reply conans/client/tools/oss.py:236]: Not a big deal though. We will have a look to this legacy code here, and probably hide or even remove some of it. I have been having a look at conan-center-index, and it is not very clear what is the usage of the Windows version detection (if the Windows version is relevant, it should probably go in a subsetting, but not automatically detected by a recipe. This is the criteria we are following for 2.0)

File: conans/client/tools/oss.py [hunk 5]
```diff
@@ -275,9 +277,9 @@ def get_freebsd_version():
     def get_solaris_version_name(version):
         if not version:
             return None
-        elif version.minor() == "5.10":
+        elif version.minor == "10":
```
Comment (lasote) [inline conans/client/tools/oss.py]: Shouldn't we keep the major just in case the "6" comes?

**Review by lasote — APPROVED** [review]
Great!

File: conans/client/tools/oss.py (hunk 5 above)
Comment (memsharded) [reply conans/client/tools/oss.py]: Do you really think that a Solaris major version is going to be released? 😄 

And if it is released, still a fix would be necessary to add the new model anyway. But sure, no problem, I can add a check on the major too.

Comment (memsharded) [reply conans/client/tools/oss.py]: Fixed, added 5 major check
